framework/src/auxkernels/AuxKernel.C

```diff
@@ -108,6 +111,12 @@ AuxKernel::AuxKernel(const InputParameters & parameters) :
   for (const auto & it : coupled_vars)
     for (const auto & var : it.second)
       _depend_vars.insert(var->name());
+
+  if (_displaced_var)
+  {
+    addMooseVariableDependency(_displaced_var);
```

Comment:
Actually - now that I think about it... we have issues with updating the
Gradient of coupled AuxVariables in-between AuxKernel evaluation…

We actually need to re-evaluate those... and they will be different on the
displaced mesh.

Ugh.

On Mon, Jun 6, 2016 at 11:20 AM Yaqi <notifications@github.com> wrote:

> In framework/src/auxkernels/AuxKernel.C
> <https://github.com/idaholab/moose/pull/7151#discussion_r65909743>:
>
> > @@ -108,6 +111,12 @@ AuxKernel::AuxKernel(const InputParameters & parameters) :
> >    for (const auto & it : coupled_vars)
> >      for (const auto & var : it.second)
> >        _depend_vars.insert(var->name());
> > +
> > +  if (_displaced_var)
> > +  {
> > +    addMooseVariableDependency(_displaced_var);
>
> I actually mean that if we can resize necessary arrays somewhere, we can
> get rid of most of extra codes because we are essentially just writing in
> AuxKernel, there is indeed no dependency on the moose variable on the
> displaced mesh.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7151/files/63ee705db72c0e8b66427760050fbdd412f0af59#r65909743>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1JMUKpFLt4HXRMwySMATyt7yv9Byo1ks5qJDq1gaJpZM4ItxmV>
> .
>